tests/test_parsers.py

```diff
@@ -403,10 +393,57 @@ def test_round_trip(self):
             "skos": "http://www.w3.org/2004/02/skos/core#",
             "sssom": "https://w3id.org/sssom/",
         }
+        self.assertEqual(expected_bimap, msdf.converter.bimap)
+
+        with TemporaryDirectory() as directory:
+            path = Path(directory).joinpath("test.sssom.x")
+            with path.open("w") as file:
+                write_func(msdf, file)
+
+            reconstituted_msdf = parse_func(path)
+            reconstituted_msdf.clean_prefix_map(strict=True)
+
+            test_meta = {
+                "mapping_set_title": "A title",
+                "license": "https://w3id.org/sssom/license/test",
+            }
+            reconstituted_msdf_with_meta = parse_func(path, meta=test_meta)
+            reconstituted_msdf_with_meta.clean_prefix_map(strict=True)
+
+        # Ensure the prefix maps are equal after json parsing and cleaning
         self.assertEqual(
-            expected_prefix_map,
-            read_metadata[CURIE_MAP],
+            set(expected_bimap),
+            set(reconstituted_msdf.prefix_map),
+            msg="Reconstituted prefix map has different CURIE prefixes",
         )
+        self.assertEqual(
+            expected_bimap,
+            reconstituted_msdf.prefix_map,
+            msg="Reconstituted prefix map has different URI prefixes",
+        )
+
+        # Ensure the shape, labels, and values in the data frame are the same after json parsing and cleaning
+        self.assertTrue(msdf.df.equals(reconstituted_msdf.df))
+
+        # Ensure the metadata is the same after json parsing and cleaning
+        self.assertEqual(msdf.metadata, reconstituted_msdf.metadata)
+
+        combine_meta = dict(ChainMap(msdf.metadata, test_meta))
+
+        # Ensure the metadata after json parsing with additional metadata corresponds to
+        # a chain of the original metadata with the test metadata.
+        # In particular, this ensures that fields in the test metadata provided are added
+        # to the MappingSet if they are not present, but not updated if they are already present.
+        self.assertEqual(combine_meta, reconstituted_msdf_with_meta.metadata)
+
+    def test_round_trip_json(self):
+        """Test writing then reading JSON."""
+        self._basic_round_trip("json")
+
+    def test_round_trip_rdf(self):
```

Comment:
currently failing because no ORCID prefix is preserved during round-trip

Reply:
Fixed